libavpipe/src/avpipe_format.c

```diff
@@ -132,6 +136,46 @@ selected_decoded_audio(
     return -1;
 }
 
+/*
+ * Return the output stream index for a given audio xc_param audio_index.
+ * The source has one or multiple audio streams - they each have a 'stream_index'
+ * The xc_params audio_index specifies which of these audio streams are selected for encoding. This is a
+ * contiguous array with each element specifying the source stream_index, eg:
+ *   audio_index[0] = 1;
+ *   audio_index[1] = 3;
+ *   audio_index[2] = 4;
+ * This function takes the index into the audio_index params (eg. 0, 1, 2 above). This is often the output of selected_decoded_audio()
+ * The convention for the encoder (output) audio_stream_index[] storage is different than the input:
+ * - for audio join/merge/pan always store the audio at index 0
+ * - for VOD ad live ingest:
+ *   audio_stream_index[0] = unset
+ *   audio_stream_index[1] = 1
+ *   audio_stream_index[2] = unset
+ *   audio_stream_index[3] = 3
+ *   audio_stream_index[4] = 4
+ */
+int
+audio_output_stream_index(
```

Comment:
I see this comment in the `coderctx_t` declaration:
```c
int audio_stream_index[MAX_STREAMS];                /* Audio input stream indexes */
```

I think we should be more clear what "input" and "output" mean, e.g. "encoder_output". It's hard to tell what `audio_output_stream_index` means